src/NuGet.Clients/NuGet.SolutionRestoreManager/ISolutionRestoreChecker.cs

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+using NuGet.Commands;
+using NuGet.ProjectModel;
+
+namespace NuGet.SolutionRestoreManager
+{
+    /// <summary>
+    /// An up to date checker for a solution.
+    /// </summary>
+    public interface ISolutionRestoreChecker
+    {
+        /// <summary>
+        /// Given the current dependency graph spec, perform a fast up to date check and return the dirty projects.
+        /// The checker itself caches the DependencyGraphSpec it is provided & the last restore status, reported through <see cref="ReportStatus(IReadOnlyList{RestoreSummary})"/>.
```

Comment:
Can you check intellisense to make sure the `&` works correctly? XMLDoc is XML, so I'd expect this needs to be `&amp;` instead.

Reply:
I guess depends how you define works correctly :D
I'll change it to the word `and`